manifests/charts/istio-control/istio-discovery/templates/deployment.yaml

```diff
@@ -151,6 +151,10 @@ spec:
             value: "{{ .Values.global.istiod.enableAnalysis }}"
           - name: CLUSTER_ID
             value: "{{ $.Values.global.multiCluster.clusterName | default `Kubernetes` }}"
+{{- if not (eq .Values.global.remotePilotAddress "") }}
+          - name: ISTIOD_CUSTOM_HOST
+            value: istiod{{- if not (eq .Values.revision "") }}-{{ .Values.revision }}{{- end }}
```

Comment:
donot we have same value in L149, and it is already signed in the certificate?

Reply:
In this multi-cluster scenario, remote cluster normally has different host with primary cluster, so it's necessary to add customHost to server's dns names. With the bootstrap logic of pilot, this value is setted by env ISTIOD_CUSTOM_HOST.
Also, I think ISTIOD_ADDR in L149 is not used in code, so ISTIOD_CUSTOM_HOST is necessary here.